docs/caching.md

```diff
@@ -0,0 +1,238 @@
+
+## Query frontend cache
+
+Query Frontend supports caching query results and reuses them on subsequent queries. If the cached results are incomplete, Query Frontend calculates the required subqueries and executes them in parallel on downstream queriers. Query Frontend can optionally align queries with their step parameter to improve the cacheability of the query results.
+
+# Configuration for memcached
+
+## Base configuration
+
+Each component can be configured with the following base configuration.
+
+```yaml
+type: MEMCACHED # Case-insensitive
+config:
+  addresses: []
+  timeout: 500ms
+  max_idle_connections: 100
+  max_async_concurrency: 20
+  max_async_buffer_size: 10000
+  max_item_size: 1MiB
+  max_get_multi_concurrency: 100
+  max_get_multi_batch_size: 0
+  dns_provider_update_interval: 10s
+```
+
+## Index cache specific for memcached
+
+The index cache has no exceptions on the base configuration
+
+## Caching bucket specific for memcached
+
+Additional options to configure various aspects of [chunks](../design.md/#chunk) cache are available:
+
+- `chunk_subrange_size`: size of segment of [chunks](../design.md/#chunk) object that is stored to the cache. This is the smallest unit that chunks cache is working with.
+- `max_chunks_get_range_requests`: how many "get range" sub-requests may cache perform to fetch missing subranges.
+- `chunk_object_attrs_ttl`: how long to keep information about [chunk file](../design.md/#chunk-file) attributes (e.g. size) in the cache.
+- `chunk_subrange_ttl`: how long to keep individual subranges in the cache.
+
+Following options are used for metadata caching (meta.json files, deletion mark files, iteration result):
+
+- `blocks_iter_ttl`: how long to cache result of iterating blocks.
+- `metafile_exists_ttl`: how long to cache information about whether meta.json or deletion mark file exists.
+- `metafile_doesnt_exist_ttl`: how long to cache information about whether meta.json or deletion mark file doesn't exist.
+- `metafile_content_ttl`: how long to cache content of meta.json and deletion mark files.
+- `metafile_max_size`: maximum size of cached meta.json and deletion mark file. Larger files are not cached.
+
+A full example would be: 
+
+
+```yaml
+type: MEMCACHED # Case-insensitive
+config:
+  addresses: []
+  timeout: 500ms
+  max_idle_connections: 100
+  max_async_concurrency: 20
+  max_async_buffer_size: 10000
+  max_item_size: 1MiB
+  max_get_multi_concurrency: 100
+  max_get_multi_batch_size: 0
+  dns_provider_update_interval: 10s
+chunk_subrange_size: 16000
+max_chunks_get_range_requests: 3
+chunk_object_attrs_ttl: 24h
+chunk_subrange_ttl: 24h
+blocks_iter_ttl: 5m
+metafile_exists_ttl: 2h
+metafile_doesnt_exist_ttl: 15m
+metafile_content_ttl: 24h
+metafile_max_size: 1MiB  
+```
+
+## Query frontend specific
+
+`expiration` specifies memcached cache valid time , If set to 0s, so using a default of 24 hours expiration time
```

Comment:
`expiration` means how long the data is stored in Memcached i.e. after that time Memcached evicts those items. Yep, the default is `24h` :hugs: I think we can improve this sentence:
```suggestion
`expiration` specifies how long memcached itself keeps items. After that time, memcached evicts those items. `0s` means the default duration of `24h`.
```